src/main/java/org/mastodon/mamut/nearest/NearestObjectStatModel.java

```diff
@@ -0,0 +1,472 @@
+package org.mastodon.mamut.nearest;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Objects;
+import java.util.function.ToDoubleBiFunction;
+
+import org.jdom2.IllegalAddException;
+import org.mastodon.feature.FeatureModel;
+import org.mastodon.feature.FeatureProjection;
+import org.mastodon.mamut.feature.SpotIntensityFeature;
+import org.mastodon.mamut.model.Spot;
+import org.mastodon.mamut.nearest.NearestObjectStatModel.NearestObjectStatItem;
+import org.mastodon.ui.coloring.feature.FeatureProjectionId;
+import org.mastodon.ui.coloring.feature.TargetType;
+import org.scijava.listeners.Listeners;
+
+import bdv.ui.settings.style.Style;
+import gnu.trove.list.array.TDoubleArrayList;
+import net.imglib2.util.Util;
+
+public class NearestObjectStatModel implements Iterable< NearestObjectStatItem >, Style< NearestObjectStatModel >
+{
+
+	public static final int MAX_N_ITEMS = 20;
+
+	public static final Collection< NearestObjectStatModel > defaults;
+
+	static
+	{
+		final NearestObjectStatModel d1 = new NearestObjectStatModel( "default" );
+		d1.add( new NearestObjectStatItem() );
+		final NearestObjectStatModel d2 = new NearestObjectStatModel( "Mean of neighbor intensities" );
+		d2.add( new NearestObjectStatItem(
+				CollectBy.SPECIFY_N,
+				6,
+				10.,
+				Value.FEATURE,
+				new FeatureProjectionId(
+						SpotIntensityFeature.KEY,
+						SpotIntensityFeature.MEAN_PROJECTION_SPEC.getKey(),
+						TargetType.VERTEX,
+						0 ),
+				Stat.MEAN,
+				false ) );
+		defaults = new ArrayList<>( 2 );
+		defaults.add( d1 );
+		defaults.add( d2 );
+	}
+
+	private final List< NearestObjectStatItem > items = new ArrayList<>();
+
+	private final Listeners.List< StatModelListener > updateListeners = new Listeners.List<>();
+
+	private String name;
+
+	public NearestObjectStatModel()
+	{}
+
+	public NearestObjectStatModel( final String name )
+	{
+		this.name = name;
+	}
+
+	@Override
+	public NearestObjectStatModel copy( final String name )
+	{
+		final NearestObjectStatModel nosm = new NearestObjectStatModel( "copy" );
+		nosm.set( this );
+		if ( name != null )
+			nosm.setName( name );
+		return nosm;
+
+	}
+
+	public void set( final NearestObjectStatModel o )
+	{
+		name = o.name;
+		items.clear();
+		for ( final NearestObjectStatItem item : o.items )
+			items.add( item );
+
+		notifyListeners();
+	}
+
+	public synchronized void setItems( final List< NearestObjectStatItem > o )
+	{
+		if ( !this.items.equals( o ) )
+		{
+			this.items.clear();
+			for ( final NearestObjectStatItem item : o )
+				this.items.add( item );
+			notifyListeners();
+		}
+	}
+
+	public List< NearestObjectStatItem > getItems()
```

Comment:
Getters and setters should be synchronized in pairs.

When one part of a getter/setter pair is synchronized the other should be too. Failure to synchronize both sides may result in inconsistent behavior at runtime as callers access an inconsistent method state.